codec-socks/src/main/java/io/netty/handler/codec/socks/SocksAuthRequestDecoder.java

```diff
@@ -28,39 +28,36 @@
  */
 public class SocksAuthRequestDecoder extends ReplayingDecoder<State> {
 
-    private SocksSubnegotiationVersion version;
-    private int fieldLength;
-    private String username;
-    private String password;
-    private SocksRequest msg = SocksCommonUtils.UNKNOWN_SOCKS_REQUEST;
-
     public SocksAuthRequestDecoder() {
         super(State.CHECK_PROTOCOL_VERSION);
     }
 
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf byteBuf, List<Object> out) throws Exception {
+        String username = null;
         switch (state()) {
             case CHECK_PROTOCOL_VERSION: {
-                version = SocksSubnegotiationVersion.valueOf(byteBuf.readByte());
+                SocksSubnegotiationVersion version = SocksSubnegotiationVersion.valueOf(byteBuf.readByte());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)